scripts/qq_official_media_probe.py

```
@@ -0,0 +1,644 @@
+            urls.extend(_iter_json_urls(item))
+        return urls
+    return []
+
+
+def _extract_xml_urls(raw_xml: str | None) -> list[str]:
+    if not raw_xml:
+        return []
+
+    urls: list[str] = []
+    try:
+        root = ElementTree.fromstring(raw_xml)
```

Ruff (0.15.20)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)